src/Jackalope/Node.php

```diff
@@ -942,12 +959,12 @@ public function getMixinNodeTypes()
     {
         $this->checkState();
 
-        if (!isset($this->properties['jcr:mixinTypes'])) {
+        if (!$this->hasProperty('jcr:mixinTypes')) {
             return array();
         }
         $res = array();
         $ntm = $this->session->getWorkspace()->getNodeTypeManager();
-        foreach ($this->properties['jcr:mixinTypes']->getValue() as $type) {
+        foreach ($this->getPropertyValue('jcr:mixinTypes') as $type) {
```

Comment:
One tricky part is knowing when we can call these "API" methods, as they will always check the state and break thiings if the operation is "internal" and doesn't care about the external state.